modal/_utils/task_command_router_client.py

```diff
@@ -458,7 +458,9 @@ async def sleep_and_update_delay_and_num_retries_remaining(e: Exception):
                         file_descriptor=file_descriptor,
                     )
 
-                    # Scope auth retry strictly to the initial send (where headers/auth are sent).
+                    # Auth retry is scoped to a single refresh per streaming attempt. While auth metadata is
+                    # sent on request start, UNAUTHENTICATED may sometimes surface during iteration,
+                    # so we handle it at both send and receive boundaries.
```

Comment:
Can the whole body by in the loop? i.e:

```python
try:
    await s.send_message(req, end=True)
    num_auth_retries = 0
    async for item in s:
        delay_secs = self.stream_stdio_retry_delay_secs
        offset += len(item.data)
        yield item
except GRPCError as exec:
    ...
```

`num_auth_retries` feels like a bool since the retry checks are always conditioned on `num_auth_retries < 1`.